mk_core/mk_event.c

```diff
@@ -176,6 +176,18 @@ int mk_event_channel_create(struct mk_event_loop *loop,
     return _mk_event_channel_create(ctx, r_fd, w_fd, data);
 }
 
+/* Destroy channel created to distribute signals */
```

Comment:
Considering that this is sort of a macro operation, wouldn't it make more sense to write something like this instead of platform specific versions? 

```
#ifdef _WIN32
typedef flb_sockfd_t mk_socket_t
#else
typedef int          mk_socket_t
#endif

int mk_event_channel_destroy(struct mk_event_loop *loop,
                             mk_socket_t r_fd, mk_socket_t w_fd,
                             void *data)
{
    mk_bug(!data);

    mk_event_del(loop, data);

    mk_event_closesocket(r_fd);
    mk_event_closesocket(w_fd);

    return 0;
}
```

This code does "swallow" the `mk_event_del` result code but considering that the descriptor could have been removed from the event loop by the consumer code for whatever reason people have to do weird stuff, do we want to take that as a signal that we don't have to close the file descriptors? I'm not sure tbh, I don't think so.

Reply:
We would have to rewrite the entire library the same way, IMO. While I agree that would be much better, I would not be a fan of doing it with this kind of library without extensive unit testing first.